test/integration/channel-transfer_test.go

```diff
@@ -0,0 +1,733 @@
+	})
+	BeforeEach(func() {
+		By("start robot")
+		robotRunner := networkFound.RobotRunner()
+		robotProc = ifrit.Invoke(robotRunner)
+		Eventually(robotProc.Ready(), network.EventuallyTimeout).Should(BeClosed())
+
+		By("start channel transfer")
+		channelTransferRunner := networkFound.ChannelTransferRunner()
+		channelTransferProc = ifrit.Invoke(channelTransferRunner)
+		Eventually(channelTransferProc.Ready(), network.EventuallyTimeout).Should(BeClosed())
+	})
+	AfterEach(func() {
+		By("stop robot")
+		if robotProc != nil {
+			robotProc.Signal(syscall.SIGTERM)
+			Eventually(robotProc.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+
+		By("stop channel transfer")
+		if channelTransferProc != nil {
+			channelTransferProc.Signal(syscall.SIGTERM)
+			Eventually(channelTransferProc.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+	})
+	/*
+		BeforeEach(func() {
+			By("add admin to acl")
+			client.AddUser(network, peer, network.Orderers[0], admin)
+
+			By("add user to acl")
+			user = client.NewUserFoundation()
+			client.AddUser(network, peer, network.Orderers[0], user)
+
+			By("emit tokens")
+			emitAmount := "1000"
+			client.TxInvokeWithSign(network, peer, network.Orderers[0],
+				cmn.ChannelFiat, cmn.ChannelFiat, admin,
+				"emit", "", client.NewNonceByTime().Get(), user.AddressBase58Check, emitAmount)
+
+			By("emit check")
+			client.Query(network, peer, cmn.ChannelFiat, cmn.ChannelFiat,
+				checkResult(checkBalance(emitAmount), nil),
+				"balanceOf", user.AddressBase58Check)
+
+			By("creating grpc connection")
+			clientCtx = metadata.NewOutgoingContext(context.Background(), metadata.Pairs("authorization", networkFound.ChannelTransfer.AccessToken))
+
+			transportCredentials := insecure.NewCredentials()
+			grpcAddress := networkFound.ChannelTransfer.HostAddress + ":" + strconv.FormatUint(uint64(networkFound.ChannelTransfer.Ports[cmn.GrpcPort]), 10)
+
+			var err error
+
+			conn, err = grpc.Dial(grpcAddress, grpc.WithTransportCredentials(transportCredentials))
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating channel transfer API client")
+			apiClient = cligrpc.NewAPIClient(conn)
+		})
+		AfterEach(func() {
+			err := conn.Close()
+			Expect(err).NotTo(HaveOccurred())
+		})
+	*/
+
+	It("transfer by admin test", func() {
+		By("add admin to acl")
+		client.AddUser(network, peer, network.Orderers[0], admin)
+
+		By("add user to acl")
+		user = client.NewUserFoundation()
+		client.AddUser(network, peer, network.Orderers[0], user)
+
+		By("emit tokens")
+		emitAmount := "1000"
+		client.TxInvokeWithSign(network, peer, network.Orderers[0],
+			cmn.ChannelFiat, cmn.ChannelFiat, admin,
+			"emit", "", client.NewNonceByTime().Get(), user.AddressBase58Check, emitAmount)
+
+		By("emit check")
+		client.Query(network, peer, cmn.ChannelFiat, cmn.ChannelFiat,
+			checkResult(checkBalance(emitAmount), nil),
+			"balanceOf", user.AddressBase58Check)
```

Comment:
It seems to me that this shared code could be put in a separate BefoeEach section?